rasa/nlu/classifiers/embedding_intent_classifier.py

```diff
@@ -823,17 +827,25 @@ def predict_label(
 
         # if X contains all zeros do not predict some label
         if label_ids.size > 0:
-            label = {
-                "name": self.inverted_label_dict[label_ids[0]],
-                "confidence": message_sim[0],
-            }
 
+            # normalise scores if turned on
+            if self.loss_type == "softmax" and self.ranking_length > 0:
+                ranking_length = self.ranking_length
+                label_ids = label_ids[:ranking_length]
+                message_sim = message_sim[:ranking_length] / (
+                    np.sum(message_sim[:ranking_length])
+                )
```

Comment:
let's just normalization function from core policy and normalize whole `message_sim`

Reply:
its is very different from the other one because it needs to truncate the length of the scores directly, while the other doesn't have them ranked and doesn't truncate it. the only thing shared is the dividing by the sum, which isn't that novel of a thing to be it's own method

Reply:
I think it's better to separate out truncation and normalization

Reply:
but you didn't want to ouput 10 when we normalized over 5 😄

Reply:
I see your suggestion for how to handle, makes sense